pkg/utils/time_utils_test.go

```diff
@@ -12,11 +12,11 @@ func TestClock(t *testing.T) {
 	t1 := time.Now()
 	t2 := clk.Now()
 
-	assert.True(t, t2.After(t1))
+	assert.True(t, t2.Compare(t1) >= 0)
```

Comment:
It's worth trying 

```suggestion
	assert.GreaterOrEqual(t, t2, t1)
```

https://pkg.go.dev/github.com/stretchr/testify/assert#GreaterOrEqualf

As you can see here, testify supports time.Time

https://github.com/stretchr/testify/blob/v1.9.0/assert%2Fassertion_compare.go#L317-L329

The output will be way better in case of error.

You can simulate it by inverting t1 and t2 temporarily and locally